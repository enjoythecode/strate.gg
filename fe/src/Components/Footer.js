import React from "react";
import { EnableSoundToggle } from "../Store/Components/EnableSoundToggle";

const Footer = () => {
  return (
    <footer>
      <p>
        <EnableSoundToggle></EnableSoundToggle>
        <br></br>
        <span>
          A project by <a href="https://enjoythecode.com">Sinan</a> dedicated to
          the play and exploration of abstract strategy board games.
          <br />
          <span>
            Take a look at the{" "}
            <a href="https://github.com/enjoythecode/strate.gg">source code.</a>
          </span>
        </span>
      </p>
    </footer>
  );
};

export default Footer;
