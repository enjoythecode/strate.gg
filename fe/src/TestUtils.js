import { render } from "@testing-library/react";
import { RootStoreProvider, initRootStoreAndSocket } from "Store/RootStore";

export const createFreshRootStore = () => {
  return initRootStoreAndSocket();
};

export const renderComponentWithRootStore = (component, RootStore = null) => {
  if (RootStore === null) {
    RootStore = createFreshRootStore();
  }

  const { asFragment } = render(
    <RootStoreProvider store={RootStore}>{component}</RootStoreProvider>,
  );

  return asFragment();
};
