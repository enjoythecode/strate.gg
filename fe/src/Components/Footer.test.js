import { renderComponentWithRootStore } from "TestUtils";

import Footer from "Components/Footer";

describe("snapshots", () => {
  it("default footer", () => {
    const tree = renderComponentWithRootStore(<Footer />);
    expect(tree).toMatchSnapshot();
  });
});
