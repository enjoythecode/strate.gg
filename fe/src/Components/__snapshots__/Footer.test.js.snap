// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots default footer 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-h8m1q6"
  >
    <footer>
      <span>
        A project by dedicated to the play and exploration of abstract strategy board games.
        <br />
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="https://enjoythecode.com"
            >
              About the Developer
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="https://github.com/enjoythecode/strate.gg"
            >
              Source Code
            </a>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="https://lichess.org/"
            >
              Proudly inspired by lichess.org
            </a>
          </div>
        </div>
      </span>
    </footer>
  </div>
</DocumentFragment>
`;
